me.chanjar.weixin.common.error.WxErrorException: 错误代码：-99, 错误信息：平台签名验证错误

### 简要描述
小程序的即时配送接口在接入配送平台后，使用 addOrder 接口创建配送单时，提示 "me.chanjar.weixin.common.error.WxErrorException: 错误代码：-99, 错误信息：平台签名验证错误"

### 模块版本情况
* WxJava 模块名: weixin-java-miniapp
* WxJava 版本号: 4.3.6.B

### 详细描述
查看 wxMaService.post 方法的执行解析 AddOrderRequest 对象的 json 中不含有 "delivery_sign" 参数，但查看 AddOrderRequest 的基类 WxMaDeliveryBaseRequest ，发现是含有 "delivery_sign" 参数的声明的，不明白是什么原因。
WxMaDeliveryBaseRequest 类中的 delivery_sign 参数是通过 getDeliverySign 方法返回的，该方法使用 shopid、shop_order_id、appSecret 参数加密，这几个参数均已提供，但却不生成  "delivery_sign" 。

### 日志
Caused by: me.chanjar.weixin.common.error.WxErrorException: 错误代码：-99, 错误信息：平台签名验证错误
	at cn.binarywang.wx.miniapp.api.impl.WxMaImmediateDeliveryServiceImpl.parse(WxMaImmediateDeliveryServiceImpl.java:213)
	at cn.binarywang.wx.miniapp.api.impl.WxMaImmediateDeliveryServiceImpl.addOrder(WxMaImmediateDeliveryServiceImpl.java:97)

--- a/weixin-java-miniapp/src/main/java/cn/binarywang/wx/miniapp/api/impl/WxMaImmediateDeliveryServiceImpl.java
+++ b/weixin-java-miniapp/src/main/java/cn/binarywang/wx/miniapp/api/impl/WxMaImmediateDeliveryServiceImpl.java
@@ -98,6 +98,7 @@ public class WxMaImmediateDeliveryServiceImpl implements WxMaImmediateDeliverySe
    */
   @Override
   public AddOrderResponse addOrder(final AddOrderRequest request) throws WxErrorException {
+    request.getDeliverySign();
     return this.parse(this.wxMaService.post(WxMaApiUrlConstants.InstantDelivery.PlaceAnOrder.ADD_ORDER, request),
       AddOrderResponse.class);
   }
@@ -115,6 +116,7 @@ public class WxMaImmediateDeliveryServiceImpl implements WxMaImmediateDeliverySe
    */
   @Override
   public GetOrderResponse getOrder(final GetOrderRequest request) throws WxErrorException {
+    request.getDeliverySign();
     return this.parse(this.wxMaService.post(WxMaApiUrlConstants.InstantDelivery.GET_ORDER, request),
       GetOrderResponse.class);
   }
@@ -131,6 +133,7 @@ public class WxMaImmediateDeliveryServiceImpl implements WxMaImmediateDeliverySe
    */
   @Override
   public CancelOrderResponse cancelOrder(final CancelOrderRequest request) throws WxErrorException {
+    request.getDeliverySign();
     return this.parse(this.wxMaService.post(WxMaApiUrlConstants.InstantDelivery.Cancel.CANCEL_ORDER, request),
       CancelOrderResponse.class);
   }
@@ -147,6 +150,7 @@ public class WxMaImmediateDeliveryServiceImpl implements WxMaImmediateDeliverySe
    */
   @Override
   public AbnormalConfirmResponse abnormalConfirm(final AbnormalConfirmRequest request) throws WxErrorException {
+    request.getDeliverySign();
     return this.parse(this.wxMaService.post(WxMaApiUrlConstants.InstantDelivery.Cancel.ABNORMAL_CONFIRM, request),
       AbnormalConfirmResponse.class);
   }

--- a/weixin-java-miniapp/src/main/java/cn/binarywang/wx/miniapp/bean/delivery/base/WxMaDeliveryBaseRequest.java
+++ b/weixin-java-miniapp/src/main/java/cn/binarywang/wx/miniapp/bean/delivery/base/WxMaDeliveryBaseRequest.java
@@ -113,7 +113,8 @@ public abstract class WxMaDeliveryBaseRequest implements Serializable {
             str = str.concat(getShopOrderId());
         }
         str = str.concat(getAppSecret());
-        return DigestUtils.sha1Hex(str);
+        this.deliverySign = DigestUtils.sha1Hex(str);
+        return this.deliverySign;
     }
 
 }

--- a/weixin-java-miniapp/src/test/java/cn/binarywang/wx/miniapp/test/AddOrderJsonTest.java
+++ b/weixin-java-miniapp/src/test/java/cn/binarywang/wx/miniapp/test/AddOrderJsonTest.java
@@ -1,0 +1,20 @@
+package cn.binarywang.wx.miniapp.test;
+
+import cn.binarywang.wx.miniapp.bean.delivery.AddOrderRequest;
+import me.chanjar.weixin.common.util.json.WxGsonBuilder;
+import org.testng.annotations.Test;
+
+public class AddOrderJsonTest {
+
+  /**
+   * 验证转化Json时是否有deliverySign
+   */
+  @Test
+  public void test(){
+    AddOrderRequest request = new AddOrderRequest();
+    request.setShopId("1");
+    request.setAppSecret("2");
+    request.getDeliverySign();
+    System.out.printf(WxGsonBuilder.create().toJson(request));
+  }
+}

